apps/src/tutorialExplorer/image.jsx

```diff
@@ -25,12 +30,25 @@ const Image = React.createClass({
   },
 
   render() {
-    const rootClassName = 'image ' + (this.state.loaded ? 'image-loaded' : '');
+    let style;
+    if (this.state.loaded) {
+      style = {
+        opacity: 1,
+        transition: "opacity 200ms ease-in"
+      };
+    } else {
+      style = {
+        opacity: 0.1
+      };
+    }
+
+    const styleProps = Object.assign({}, this.props.style, style);
+
     return (
       <img
         ref="img"
-        {...this.props}
-        className={rootClassName}
+        src={this.props.src}
+        style={styleProps}
         onLoad={this.onImageLoad}
       />
     );
```

Comment:
@Bjvanminnen I did like the older technique of just passing in `{...this.props}` so that we don't have to be explicit about which props are provided, but I couldn't think of a good way to merge in additional style information.  Ideas for something more clever?